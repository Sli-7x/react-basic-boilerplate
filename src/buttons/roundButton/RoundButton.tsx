import * as React from 'react';
import { Content, Button } from './RoundButton.style';
import { Icon } from '../../index';

export interface IRoundButtonProps {
  onClick?: () => void;
  caption: string;
  iconname: string;
  selected: boolean;
}

export const RoundButton: React.StatelessComponent<IRoundButtonProps> = ({ onClick, caption, iconname, selected }) => (
  <Content>
    <Button onClick={onClick} selected={selected}>
      <Icon
        key={caption}
        title={caption}
        style={{ display: 'block', fontSize: '25px', color: selected ? '#fff' : '#327f85' }}
        name={iconname}
      />
    </Button>
  </Content>
);
