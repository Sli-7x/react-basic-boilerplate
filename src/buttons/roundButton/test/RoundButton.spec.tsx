import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { IRoundButtonProps, RoundButton } from '../RoundButton';

describe('RoundButton', () => {
  const props: IRoundButtonProps = {
    caption: '',
    iconname: '',
    selected: false,
  };

  it('should render correctly', () => {
    const tree = renderer.create(<RoundButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();

    const tree2 = renderer.create(<RoundButton {...props} selected={true} />).toJSON();
    expect(tree2).toMatchSnapshot();
  });
});
