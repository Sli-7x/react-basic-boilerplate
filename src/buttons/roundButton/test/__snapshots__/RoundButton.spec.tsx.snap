// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div
  className="sc-bdVaJa hhLoUU"
>
  <button
    className="sc-bwzfXH iURKpz"
    selected={false}
  >
    <i
      className="material-icons"
      style={
        Object {
          "color": "#327f85",
          "display": "block",
          "fontSize": "25px",
        }
      }
      title=""
    >
      
    </i>
  </button>
</div>
`;

exports[`RoundButton should render correctly 1`] = `
<div
  className="sc-bdVaJa hhLoUU"
>
  <button
    className="sc-bwzfXH iURKpz"
    selected={false}
  >
    <i
      className="material-icons"
      style={
        Object {
          "color": "#327f85",
          "display": "block",
          "fontSize": "25px",
        }
      }
      title=""
    >
      
    </i>
  </button>
</div>
`;

exports[`RoundButton should render correctly 2`] = `
<div
  className="sc-bdVaJa hhLoUU"
>
  <button
    className="sc-bwzfXH jFYXBd"
    selected={true}
  >
    <i
      className="material-icons"
      style={
        Object {
          "color": "#fff",
          "display": "block",
          "fontSize": "25px",
        }
      }
      title=""
    >
      
    </i>
  </button>
</div>
`;
